api/tests/utils/mod.rs

```diff
@@ -125,3 +143,21 @@ impl QueryRunner {
         (to_json(data), errors.into_iter().map(to_json).collect())
     }
 }
+
+impl Drop for QueryRunner {
+    fn drop(&mut self) {
+        // If the test wasn't inside a transaction, then whatever DB changes it
+        // made will still be around - we want to clean those up. Ideally we
+        // truncate all tables here, but that sounds like a lot of work that I
+        // don't wanna do so just sticking with users for now.
+        let conn = self.db_conn();
+        if (conn.transaction_manager() as &dyn TransactionManager<PgConnection>)
+            .get_transaction_depth()
+            == 0
+        {
+            // TODO clean all tables here
+            diesel::delete(user_providers::table).execute(conn).unwrap();
+            diesel::delete(users::table).execute(conn).unwrap();
```

Comment:
This is just me spit balling but could we do some derive macro jank on all disel tables that would add them to a lazy static list or something so we could iter over all of them to delete?

Reply:
Yeah I think that would work, we would just have to remember to include that derive on the model when we add a new table. Writing a derive macro is also kind of a bitch. Tbh I think I'd rather just maintain a manual list, but I'd think about it more.

Reply:
yea it would be ass so just something to keep in mind